xpdacq/beamtime.py

```diff
@@ -244,6 +244,13 @@ def _clean_info(obj):
     """ stringtify and replace space"""
     return str(obj).strip().replace(' ', '_')
 
+class MDOrderedDict(OrderedDict):
```

Comment:
You might want to call it something different. The issue being addressed here, that python has overloaded `[bla]` to both get index `bla` from list and get value associated with key `bla` and that the dictionary usage took precedence, is a bit more generic than simply for metadata. Would it be possible name this something along the lines of `SliceableOrderedDict` and have a `get_slice` method? I think this may head off some of the unique syntax issues (it may even be a nice thing for other developers to use).